web-ui/src/main/resources/catalog/components/search/map/MapFieldDirective.js

```diff
@@ -134,20 +134,253 @@
               var updateField = function(geom) {
                 feature.setGeometry(geom);
 
-                // Write the extent as 4326 WKT polygon
-                var lonlatFeat, writer, wkt;
+                // Get projection code of current map view
+                var proj = scope.map.getView().getProjection().getCode();
+
+                /**
+                 * Transforms the given longitude-latitude pair to XY coordinates in the current projection.
+                 * @param {number} lon 
+                 * @param {number} lat 
+                 */
+                var getXY = function(lon, lat) {
```

Comment:
I think all those utilities should not be added here, but in another service. This adds way too much code inside a simple scope method.

Reply:
You've got a valid point there. Any suggestions where to put it?

Reply:
Well, there is a `GeometryService.js` file which could be a good place for these. You'd then have to import `GeometryService` in this directive.

Reply:
Another option is adding a service right next to the directive, e.g. `MapFieldDirective`. Maybe better as it keeps the code close.

Reply:
You mean `MapFieldService`, I guess? If we won't intend to use the functionality anywhere else in the future (e.g. for the main map), that probably makes sense yes.